### ISSUE TITLE
feature request: clipboard watcher

### ISSUE THREAD

**Issue by agzam** [Hammerspoon/hammerspoon#2269]
How difficult would it be to be able to "watch" the clipboard, so one can take an action, for example if something that starts with "https://" gets copied?

**Comment by latenitefilms** [comment]
There’s sadly no Apple API for pasteboard notifications, but you can use a hs.timer to check for pasteboard changes on a regular interval.

**Comment by anshukla** [comment]
Polling may inadvertently trigger continuity pasteboard, though.

**Comment by latenitefilms** [comment]
Have a look at that example code. You can use hs.pasteboard.changeCount() to see if there’s new items on the Pasteboard.

**Comment by anshukla** [comment]
Right, but I think this is the line which would cause a modal universal clipboard window to appear without user interaction if they copied an item on a different device: ``local contents = pasteboard.readString()``.

**Comment by cmsj** [comment]
@anshukla polling is unavoidable, there is no other way to watch the macOS clipboard.

I do think we probably need to pull this into the core of Hammerspoon though - there are already at least two Spoons doing clipboard watching.
